src/components/StoreDeployBlock.js

```diff
@@ -3,33 +3,45 @@ import PropTypes from "prop-types";
 import HashColor from "./UI/HashColor";
 import Input from "./UI/Input";
 import { OrangeButton } from "./UI/Button";
+import { isEmpty } from "../components/utils";
 
 class StoreDeployBlock extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      issuerName: ""
+      issuerName: "",
+      issuerNameIsValid: true
     };
-
     this.onNameChange = this.onNameChange.bind(this);
-    this.onDeployClick = this.onDeployClick.bind(this);
   }
 
   onNameChange(event) {
     this.setState({
-      issuerName: event.target.value
-    });
-  }
-
-  onDeployClick() {
-    const { adminAddress, handleStoreDeploy } = this.props;
-    handleStoreDeploy({
-      fromAddress: adminAddress,
-      name: this.state.issuerName
+      issuerName: event.target.value,
+      issuerIsValid: isEmpty(event.target.value)
     });
   }
 
   render() {
+    const inputMessage = this.state.issuerIsValid
+      ? "Issuer Name cannot be empty."
+      : "";
+
+    const onDeployClick = () => {
+      const { adminAddress, handleStoreDeploy } = this.props;
+      const { issuerName, issuerNameIsValid } = this.state;
+      if (!isEmpty(issuerName) && issuerNameIsValid) {
```

Comment:
Actually the state `issuerNameIsValid` is not exactly used here. Will it make more sense for it to track if that field has been updated before? Since you do the checks here. If not people will wonder why we check isEmpty here twice in this statement.

Reply:
Also, why not put onDeployClick in the class method instead of nesting it here?

Reply:
Consider `!issuerName`